lib/ts/BufferWriterFormat.cc

```diff
@@ -403,6 +404,107 @@ namespace bw_fmt
     return w;
   }
 
+  /// Format for floating point values. Seperates floating point into a whole number and a
+  /// fraction. The fraction is converted into an unsigned integer based on the specified
+  /// precision, spec._prec. ie. 3.1415 with precision two is seperated into two unsigned
+  /// integers 3 and 14. The different pieces are assembled and placed into the BufferWriter.
+  /// The default is two decimal places. ie. X.XX. The value is always written in base 10.
+  ///
+  /// format: whole.fraction
+  ///     or: left.right
+  BufferWriter &
+  Format_Floating(BufferWriter &w, BWFSpec const &spec, double f, bool neg_p)
+  {
+    if (f == static_cast<uint64_t>(f)) { // integral
```

Comment:
If you're doing this, you need to cache it
```
uint64_t i = static_cast<uint64_t>(f);
if (i == f)  return Format_Integer(..., f, neg_p);
```

Reply:
done.